modules/representation_management/app/models/representation_management/accredited_entity_query.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module RepresentationManagement
+  class AccreditedEntityQuery
+    include ActiveModel::Model
+
+    MAXIMUM_RESULT_COUNT = 10
+    WORD_SIMILARITY_THRESHOLD = 0.7
+
+    def initialize(query_string)
+      @query_string = query_string
+    end
+
+    # rubocop:disable Metrics/MethodLength
+    def results
+      return [] if @query_string.blank?
+
+      sql = <<-SQL.squish
+        WITH combined AS (
+          SELECT
+            id,
+            'AccreditedIndividual' AS model_type,
+            full_name AS name,
+            levenshtein(full_name, ?) AS distance
+          FROM
+            accredited_individuals
+          WHERE
+            word_similarity(?, full_name) >= ?
+            AND location IS NOT NULL
+
+          UNION ALL
+
+          SELECT
+            id,
+            'AccreditedOrganization' AS model_type,
+            name AS name,
+            levenshtein(name, ?) AS distance
+          FROM
+            accredited_organizations
+          WHERE
+            word_similarity(?, name) >= ?
+            AND location IS NOT NULL
+        )
+        SELECT
+          id,
+          model_type,
+          distance
+        FROM
+          combined
+        ORDER BY
+          distance ASC
+        LIMIT
+          ?;
+      SQL
+
+      array_results = ActiveRecord::Base.connection.exec_query(
+        ActiveRecord::Base.send(:sanitize_sql_array, [
```

Comment:
You should be able to dry this up a little with named placeholders in the query: `:query_string` etc

Reply:
Done!